cli/src/commands/init.rs

```diff
@@ -21,61 +23,78 @@ pub async fn handle_init(channel: ApiChannel) -> Result<bool, WKCliError> {
-            refresh_token_loader.set_message("Refreshing tokens...");
-
-            let updated_config = match auth::okta::refresh_tokens(&config).await {
-                Ok(new_tokens) => {
-                    current_config.auth = Some(new_tokens.into());
-                    refresh_token_loader.finish_and_clear();
-
-                    current_config
-                }
-                Err(err) => {
-                    refresh_token_loader.finish_and_clear();
-                    match err {
-                        WKCliError::AuthError(AuthError::OktaRefreshTokenExpired { .. }) => {
-                            eprintln!("The refresh token is expired. You have to login again.");
-                            login_and_create_config(current_config).await?
-                        }
-                        err => return Err(err),
-                    }
-                }
-            };
-
-            current_config = updated_config;
-        } else {
-            colored_println!("You are logged in as: {}.\n", login_selections[selection]);
-        }
-
-        current_config
-    };
+    let mut new_config = Config::load_from_default_path()?;
+    new_config = handle_application(new_config, channel).await?;
+    new_config = handle_gcloud_auth(new_config).await?;
+    new_config = handle_bunker_auth(new_config).await?;
+
+    colored_println!(
+        r#"
+    Your Wukong CLI is configured and ready to use!
+
+    * Commands that require authentication will use {} by default
+    * Commands will reference application {} by default
+    Run `wukong config help` to learn how to change individual settings
+
+    Some things to try next:
+
+    * Run `wukong --help` to see the wukong command groups you can interact with. And run `wukong COMMAND help` to get help on any wukong command.
+                             "#,
+        new_config.auth.okta.as_ref().unwrap().account,
+        new_config.core.application
+    );
+
+    new_config
+        .save_to_default_path()
+        .expect("Config file save failed");
+
+    Ok(true)
+}
+
+async fn handle_bunker_auth(mut config: Config) -> Result<Config, WKCliError> {
+    let agree_to_authenticate = Confirm::with_theme(&ColorfulTheme::default())
+        .with_prompt(format!(
+            "{} {}",
+            "(Optional)".bright_black(),
+            "Do you want to authenticate against Bunker? You may do it later when neccessary"
+        ))
+        .default(true)
+        .interact()?;
+
+    if agree_to_authenticate {
+        vault::get_token_or_login(&mut config).await?;
+    }
+
+    Ok(config)
+}
 
+async fn handle_gcloud_auth(config: Config) -> Result<Config, WKCliError> {
+    let agree_to_authenticate = Confirm::with_theme(&ColorfulTheme::default())
+        .with_prompt(format!(
+            "{} {}",
+            "(Optional)".bright_black(),
+            "Do you want to authenticate against Google Cloud? You may do it later when neccessary"
+        ))
+        .default(false)
```

Comment:
But the default value for google login is `false`, unless you have a reason why they are different, I think we should make them consistent, either both `true` or both `false`.